pkg/client/command_line.go

```diff
@@ -15,62 +16,215 @@ func AddArmadaApiConnectionCommandlineArgs(rootCmd *cobra.Command) {
 	viper.BindPFlag("armadaUrl", rootCmd.PersistentFlags().Lookup("armadaUrl"))
 }
 
-func LoadCommandlineArgsFromConfigFile(cfgFile string) {
+// LoadCommandlineArgsFromConfigFile load config from at exePath/armadactl-defaults.yaml, where
+// exePath is the path to the armadactl executable, the provided cfgFile, or, if it cfgFile="",
+// $HOME/.armadactl
+func LoadCommandlineArgsFromConfigFile(cfgFile string) error {
+
+	// read config at exePath/armadactl-defaults.yaml, where exePath is the path to the armadactl
+	// executable (if it exists)
 	exePath, err := os.Executable()
 	if err != nil {
+		return fmt.Errorf("[LoadCommandlineArgsFromConfigFile] error finding executable path: %s", err)
+	}
 
-		log.Errorf("Can't find executable path because %s\n", err)
-		os.Exit(1)
-	} else {
-		exeDir := filepath.Dir(exePath)
-		viper.SetConfigFile(exeDir + "/armadactl-defaults.yaml")
-		err := viper.ReadInConfig()
-		if err != nil {
-			switch err.(type) {
-			case viper.ConfigFileNotFoundError:
-			case *os.PathError:
-				// No default config is fine
-			default:
-				log.Errorf("Can't read config file %s because %s\n", viper.ConfigFileUsed(), err)
-				os.Exit(1)
-			}
+	exeDir := filepath.Dir(exePath)
+	configPath := filepath.Join(exeDir, "/armadactl-defaults.yaml")
+	viper.SetConfigFile(configPath)
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			// Config file not found; ignore
+		} else {
+			return fmt.Errorf("[LoadCommandlineArgsFromConfigFile] error reading config file %s: %s", viper.ConfigFileUsed(), err)
 		}
 	}
 
-	if cfgFile != "" {
-		// Use config file from the flag.
+	// if no cfgFile is provided, use $HOME/.armadactl
+	if len(cfgFile) > 0 {
 		viper.SetConfigFile(cfgFile)
 	} else {
-		// Find home directory.
-		home, err := homedir.Dir()
+		homeDir, err := os.UserHomeDir()
 		if err != nil {
-			log.Error(err)
-			os.Exit(1)
+			return fmt.Errorf("[LoadCommandlineArgsFromConfigFile] error getting user home directory: %s", err)
 		}
-
-		viper.AddConfigPath(home)
+		viper.AddConfigPath(homeDir)
 		viper.SetConfigName(".armadactl")
 	}
 
-	viper.AutomaticEnv() // read in environment variables that match
+	// read environment variables
+	viper.AutomaticEnv()
 
-	// If a config file is found, read it in.
-	err = viper.MergeInConfig()
-
-	if err != nil {
-		switch err.(type) {
-		case viper.ConfigFileNotFoundError:
-			// This only occurs when looking for the default .armadactl file and it is not present
-			// This is not an error as users don't have to specify it, so do nothing
-		default:
-			log.Errorf("Can't read config file %s because %s\n", viper.ConfigFileUsed(), err)
-			os.Exit(1)
+	// merge in new config with those loaded from armadactl-defaults.yaml
+	// (note the call to viper.MergeInConfig instead of viper.ReadInConfig)
+	if err := viper.MergeInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			// Config file not found; ignore
```

Comment:
again I'd log this

Reply:
See response to previous comment on this.